src/rsvc/srv_common.c

```diff
@@ -239,11 +249,8 @@ ds_rsvc_get_attr(struct ds_rsvc *svc, struct rdb_tx *tx, rdb_path_t *path,
 
 		if (rc == -DER_NONEXIST) {
 			/* attribute do not exist */
-			iovs[j].iov_buf_len = sizes[i];
-			iovs[j].iov_len = 0;
+			iovs[j - 1].iov_buf_len += sizes[i];
```

Comment:
Sorry, I'm not confident I understand the intention of this change. Is this change correct if the first key is nonexistent (i.e., if j == 1 && i == 0), in which case we will be modifying the sizes array iov.

Reply:
OK, seems correct. Thanks.